libevm/EVMC.cpp

```diff
@@ -23,33 +23,38 @@ EVM::EVM(evm_instance* _instance) noexcept : m_instance(_instance)
 
 owning_bytes_ref EVMC::exec(u256& io_gas, ExtVMFace& _ext, OnOpFunc const& _onOp)
 {
-	bool rejected = false;
-	// TODO: Rejecting transactions with gas limit > 2^63 can be used by attacker to take JIT out of scope
-	rejected |= io_gas > std::numeric_limits<int64_t>::max(); // Do not accept requests with gas > 2^63 (int64 max)
-	rejected |= _ext.envInfo().number() > std::numeric_limits<int64_t>::max();
-	rejected |= _ext.envInfo().timestamp() > std::numeric_limits<int64_t>::max();
-	rejected |= _ext.envInfo().gasLimit() > std::numeric_limits<int64_t>::max();
-	if (rejected)
-	{
-		cwarn << "Execution rejected by EVM JIT (gas limit: " << io_gas << "), executing with interpreter";
-		return VMFactory::create(VMKind::Interpreter)->exec(io_gas, _ext, _onOp);
-	}
+    bool rejected = false;
+    // Do not accept requests with gas > 2^63 (int64 max)
+    rejected |= io_gas > std::numeric_limits<int64_t>::max();
+    rejected |= _ext.envInfo().number() > std::numeric_limits<int64_t>::max();
+    rejected |= _ext.envInfo().timestamp() > std::numeric_limits<int64_t>::max();
+    rejected |= _ext.envInfo().gasLimit() > std::numeric_limits<int64_t>::max();
```

Comment:
Would be nice having an option which disables the fallback to interpreter if these conditions fail.

Reply:
They are quite legacy one, affected only by some exotic tests.

Reply:
That is fair enough, but I'd like to be sure that it is possible running a pure eWASM client without silently falling back on the interpreter.

Reply:
You will not be able to run eWASM anyway, because these values will not fit into 64-bit types in EVM-C.
I can change them into failures, let's see what tests are there.

Reply:
Yep, the option would mean to throw a failure here.

I guess you want this configurable in order to successfully run state tests using evmjit.